server/controllers/recipes.js

```diff
@@ -94,73 +113,85 @@ export default {
       })
       .then(favoriteRecipeList => res.status(200).json({
         favRecipes: favoriteRecipeList.rows
-      }))
-      .catch(() => res.status(422).json({ error: 'Fav Recipe Not Created yet' }));
+      }));
   },
 
-  update(req, res) {
-    const { userDetail } = req.decoded;
+  /**
+   * function updateRecipe
+   * @param {any} req
+   * @param {any} res
+   * @returns {object} updated recipe
+   */
+  updateRecipe(req, res) {
+    const { id } = req.decoded;
+    const {
+      recipeName,
+      description,
+      ingredients,
+      imageUrl,
+      preptime,
+      servings
+    } = req.body;
     return Recipe
       .find({
         where: {
           id: req.params.recipeId,
         },
-      }).then((Recipefound) => {
-        if (Recipefound && Recipefound.userId === userDetail.id) {
-          return Recipefound
+      })
+      .then((recipeFound) => {
+        if (!recipeFound) {
+          res.status(404).send({ error: 'Recipe not found' });
+        }
+        if (recipeFound.userId === id) {
+          recipeFound
             .update({
-              recipeName: req.body.recipeName || Recipefound.recipeName,
-              description: req.body.description || Recipefound.description,
-              ingredients: req.body.ingredients || Recipefound.ingredients,
-              imageUrl: req.body.imageUrl || Recipefound.imageUrl,
-              preptime: req.body.preptime || Recipefound.preptime,
-              servings: req.body.servings || Recipefound.servings,
-            }, {
-              where: {
-                id: req.params.recipeId
-              }
+              recipeName: recipeName || recipeFound.recipeName,
```

Comment:
Not a blocker, but I think this process of checking if a field should be updated can be improved, not sure how right now.